api/app_analytics/analytics_db_service.py

```diff
@@ -14,34 +14,103 @@
     FeatureEvaluationBucket,
     Resource,
 )
+from dateutil.relativedelta import relativedelta
 from django.conf import settings
 from django.db.models import Sum
 from django.utils import timezone
 
 from environments.models import Environment
 from features.models import Feature
+from organisations.models import Organisation
 
-ANALYTICS_READ_BUCKET_SIZE = 15
+from . import constants
+from .typing import PERIOD_TYPE
 
 
 def get_usage_data(
-    organisation, environment_id=None, project_id=None
-) -> List[UsageData]:
+    organisation: Organisation,
+    environment_id: int | None = None,
+    project_id: int | None = None,
+    period: PERIOD_TYPE | None = None,
+) -> list[UsageData]:
+    now = timezone.now()
+
+    date_stop = date_start = None
+    period_starts_at = period_ends_at = None
+
+    match period:
+        case constants.CURRENT_BILLING_PERIOD:
+            if not getattr(organisation, "subscription_information_cache", None):
```

Comment:
nit:

```suggestion
            if not organisation.subscription_information_cache.exists():
```

Reply:
As mentioned here: https://github.com/Flagsmith/flagsmith/pull/3870#discussion_r1611667665

The suggestion is incorrect